step-api/src/main/java/org/jenkinsci/plugins/workflow/steps/AbstractSynchronousNonBlockingStepExecution.java

```diff
@@ -65,7 +65,6 @@ public void stop(Throwable cause) throws Exception {
         if (task != null) {
             task.cancel(true);
         }
-        getContext().onFailure(cause);
```

Comment:
The regular abort button should call `task.cancel(true)` which ought to wind up calling `Thread.interrupt`, as `SynchronousNonBlockingStepTest` still demonstrates. The `doTerm` fallback should only be necessary if that thread ignores interrupts (or catches it but then retries its task, etc.).